src/server/rpc/rpc-manager.js

```diff
@@ -14,9 +14,55 @@ var fs = require('fs'),
     SocketManager = require('../socket-manager'),
     utils = require('../server-utils'),
     JsonToSnapList = require('./procedures/utils').jsonToSnapList ,
-    jsdocExtractor = require('./jsdoc-extractor.js'),
+    Docs = require('./jsdoc-extractor.js').Docs,
     RESERVED_FN_NAMES = require('../../common/constants').RPC.RESERVED_FN_NAMES;
 
+// TODO move this fn and maybe Docs class?
+// in: arg obj and input value
+// out: [parsedInput, msg]
+function parseArgValue(arg, input) {
+    let errorMsg;
+    const genericErr = `${arg.name} = ${input} is not a valid ${arg.type}. `;
+    if (input === '') {
+        if (!arg.isOptional) {
+            errorMsg = `${arg.name} is required.`;
+        }
+        input = undefined;
+        return [input, errorMsg];
+    }
+    // TODO add check for optional
+    switch (arg.type) {
```

Comment:
This would be better to move into a seperate file which returns an object of all the types where the keys of the object are the types and contains the casting functionality. This would allow the error message to be generated automatically rather than adding them everywhere.

Reply:
8 lines above this I have a line that generates the messages based on the arg info automatically, and then I'm appending specific messages to each type, like the range for latitude and long. However, it would be good to move this to a separate module.